src/main/java/com/parth/jwt/controller/JwtAuthenticationController.java

```diff
@@ -0,0 +1,54 @@
+package com.parth.jwt.controller;
+
+import com.parth.jwt.config.JwtHelper;
+import com.parth.jwt.config.JwtUserDetailsService;
+import com.parth.jwt.jwtModel.JwtRequest;
+import com.parth.jwt.jwtModel.JwtResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping
+public class JwtAuthenticationController {
+
+    @Autowired
+    AuthenticationManager authenticationManager;
+
+    @Autowired
+    private JwtHelper jwtHelper;
+
+    @Autowired
+    private JwtUserDetailsService jwtUserDetailsService;
+
+    @PostMapping("/login")
+    public JwtResponse login(@RequestBody JwtRequest request) {
+
+        this.doAuthenticate(request.getUsername(), request.getPassword());
+        UserDetails userDetails = jwtUserDetailsService.loadUserByUsername(request.getUsername());
+        String token = this.jwtHelper.generateToken(userDetails);
+
+        return JwtResponse.builder()
+                .jwtToken(token)
+                .username(userDetails.getUsername()).build();
+    }
+
+    private void doAuthenticate(String username, String password) {
+
+        UsernamePasswordAuthenticationToken authentication = new UsernamePasswordAuthenticationToken(username, password);
+        try {
+            authenticationManager.authenticate(authentication);
+        } catch (BadCredentialsException e) {
+            throw new BadCredentialsException(" Invalid Username or Password  !!");
+        }
+
+    }
+
+    @ExceptionHandler(BadCredentialsException.class)
```

Comment:
Either add exceptionHandler class or create a custom exception and add the @ResponseStatus annotation.
It is not a good practice to define exception response in the controller.